src/Program.ts

```diff
@@ -733,6 +733,36 @@ export class Program {
         return Promise.resolve(file.getHover(position));
     }
 
+    /**
+     * Compute code actions for the given file and range
+     */
+    public getCodeActions(pathAbsolute: string, range: Range) {
+        const codeActions = [] as CodeAction[];
+        const file = this.getFile(pathAbsolute);
+
+        this.plugins.emit('beforeGetCodeActions', file, range, codeActions);
+
+        //get code actions from the file
+        codeActions.push(
+            ...file.getCodeActions(range)
+        );
+
+        //get code actions from every scope this file is a member of
+        for (let key in this.scopes) {

```

Comment:
`for of` maybe?

Reply:
No, `this.scopes` is `Record<string, Scope>` so we need `for in` to iterate over the keys.